hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ReadableScheduleStoreImpl.java

```diff
@@ -41,8 +42,8 @@ public class ReadableScheduleStoreImpl implements ReadableScheduleStore {
             "Null states instance passed to ReadableScheduleStore constructor, possible state corruption.";
 
     private final ReadableKVState<ScheduleID, Schedule> schedulesById;
-    private final ReadableKVState<ProtoLong, ScheduleList> schedulesByExpirationSecond;
-    private final ReadableKVState<ProtoBytes, ScheduleList> schedulesByStringHash;
+    private final ReadableKVState<ProtoLong, ScheduleIdList> scheduleIdsByExpirationSecond;
+    private final ReadableKVState<ProtoBytes, Schedule> schedulesByStringHash;
```

Comment:
String hash values can and will collide (it's only a partial hash, not comprehensive).  Changing this to a single schedule will _miss_ duplications.  That's why we stored the whole schedule originally; storing a list of ID values might be acceptable as an alternative to storing the whole schedule.

Reply:
The reason is explained below.